File: 2105.给植物浇水2.py
class Solution:
    def minimumRefill(self, plants: List[int], capacityA: int, capacityB: int) -> int:
        i,j=0,len(plants)-1
        a,b=capacityA,capacityB
        ans=0
        while i<j:
            if a<plants[i]:
                ans+=1
                a=capacityA
            a-=plants[i]
            i+=1

            if b<plants[j]:
                ans+=1
                b=capacityB
            b-=plants[j]
            j-=1
        if i==j and max(a,b)<plants[i]:
            ans+=1

        return ans 

File: 2105.给植物浇水2.java
class Solution {
    public int minimumRefill(int[] plants, int capacityA, int capacityB) {
        int n=plants.length;
        int ans =0;
        int i=0;int j=n-1;
        int a=capacityA;int b=capacityB;

        while(i<j){
            
           
            if(a<plants[i]){
                ans++;
                a=capacityA;
            }
            a-=plants[i];
            i++;
           
            if(b<plants[j]){
                ans++;
                b =capacityB;
            } 
            b-=plants[j];
            j--;

        }
        if(i==j&&Math.max(a,b)<plants[i]){
            ans++;
        }
        return ans; 
    }
}
